schemas/README.md

```diff
@@ -1593,6 +2172,548 @@ Raw image data
 </tr>
 </table>
 
+## SceneEntity
+
+A visual element in a 3D scene. An entity may be composed of multiple primitives which all share the same frame of reference.
+
+<table>
+  <tr>
+    <th>field</th>
+    <th>type</th>
+    <th>description</th>
+  </tr>
+<tr>
+<td><code>timestamp</code></td>
+<td>
+
+time
+
+</td>
+<td>
+
+Timestamp of the entity
+
+</td>
+</tr>
+<tr>
+<td><code>frame_id</code></td>
+<td>
+
+string
+
+</td>
+<td>
+
+Frame of reference
+
+</td>
+</tr>
+<tr>
+<td><code>id</code></td>
+<td>
+
+string
+
+</td>
+<td>
+
+Identifier for the entity. A entity will replace any prior entity on the same topic with the same `id`.
+
+</td>
+</tr>
+<tr>
+<td><code>lifetime</code></td>
+<td>
+
+duration
+
+</td>
+<td>
+
+Length of time (relative to `timestamp`) after which the entity should be automatically removed. Zero value indicates the entity should remain visible until it is replaced or deleted.
+
+</td>
+</tr>
+<tr>
+<td><code>frame_locked</code></td>
```

Comment:
Entities always follow their frame_id. This setting indicates whether the entity should be interpolated in time using its timestamp with transforms rather than using the latest transform for the frame.

Reply:
That is not what the docs for this field says (or at least not how I interpreted it).

Can you give me an example of when I would want this feature enabled and disabled?

Reply:
Enabled: cylindrical piece of a robot arm, should move with the arm as /tf changes.
Disabled: lidar point cloud, should be interpreted as fixed in world/map frame even though the detection coordinates are in lidar frame.

Reply:
@jtbandes thanks. That helped me realize that roman and I were saying the same thing, I'm just getting tripped up on language.

Ok so going back to my original point - it feels to me like the default setting should be for entities to be "locked in time" to their own transform frame rather than the fixed frame. Do others agree? If so, we could invert this flag so that `true` means "lock to fixed frame".

Reply:
The default _is_ that entities are locked in time to their own transform frame. That's why `frame_locked` is _false_ by default. A value of _false_ means _interpolate_ (i.e. locked in time to my frame) while a value of _true_ means: ignore my timestamp and display me at the latest position of my frame relative to the scene.

Reply:
> The default _is_ that entities **are locked** in time to their own transform frame. That's why <code>frame_<b>locked</b></code><b> is _false_</b> by default.

:philosoraptor:

I don't think we all agree on what locked means here. Maybe this field needs a new name.

Reply:
I guess since we are all totally talking past each other, maybe we just keep this as-is since it at least matches the ROS conventions?

I think the ros docs actually did a better job of explaining this concept: _"If this marker should be frame-locked, i.e. retransformed into its frame every timestep"_.

Now that I thought about this some more, I actually think the default of `false` (objects stick to their original point in the fixed frame) makes sense. For any observations (sensor returns, tracked objects) they should not move with the transform frame. A value of `true` (objects move along with the transform frame) only makes sense for objects that are physically attached to that frame (e.g. mesh markers/entities), which seems like the exception rather than the rule.

Motion to close this thread.